### PR TITLE
wg_engine: fix aabb for empty shapes and long paths

### PR DESCRIPTION
**Pull request by SergeyLebedkin**
see https://github.com/thorvg/thorvg/issues/2843
fixed zero points path hendling
created a single shared instance of stoke generator in heap, instead of stack. No performace impact in this case

### CODE REVIEW

**Review by tinyjin — APPROVED** [review]
Looks good in this patch.

https://github.com/user-attachments/assets/58cdf1ab-95ac-4e80-bd3e-b8bd540bb269

**Review by mgrudzinska — COMMENTED** [review]
on mac:
../src/renderer/wg_engine/tvgWgRenderData.cpp:373:14: runtime error: nan is outside the range of representable values of type 'int'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior ../src/renderer/wg_engine/tvgWgRenderData.cpp:373:14 in 
../src/renderer/wg_engine/tvgWgRenderData.cpp:374:14: runtime error: nan is outside the range of representable values of type 'int'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior ../src/renderer/wg_engine/tvgWgRenderData.cpp:374:14 in 
../src/renderer/wg_engine/tvgWgRenderData.cpp:375:14: runtime error: nan is outside the range of representable va

**Comment by SergeyLebedkin** [comment]
> on mac: ../src/renderer/wg_engine/tvgWgRenderData.cpp:373:14: runtime error: nan is outside the range of representable values of type 'int' SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior ../src/renderer/wg_engine/tvgWgRenderData.cpp:373:14 in ../src/renderer/wg_engine/tvgWgRenderData.cpp:374:14: runtime error: nan is outside the range of representable values of type 'int' SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior ../src/renderer/wg_engine/tvgWgRenderData.cpp:374:14 in ../src/renderer/wg_engine/tvgWgRenderData.cpp:375:14: runtime error: nan is outside the range of representable va

Fixed, please check it

File: src/renderer/wg_engine/tvgWgGeometry.h [hunk 1]
```diff
@@ -373,7 +337,7 @@ struct WgVertexBufferInd {
         float cosine = dot(nrm, nrm1);
         float angle = std::acos(dot(nrm1, -nrm2));
         float miterRatio = 1.0f / (std::sin(angle) * 0.5f);
-        if (miterRatio <= miterLimit) {
+        if ((miterRatio <= miterLimit) && (!tvg::zero(cosine))) {
```
Comment (mgrudzinska) [inline src/renderer/wg_engine/tvgWgGeometry.h:340]: the same problem may be with sin, maybe check the angle value? 
also this change should be done as a separate commit I think

Comment (SergeyLebedkin) [reply src/renderer/wg_engine/tvgWgGeometry.h]: Make sence, fixed

File: src/renderer/wg_engine/tvgWgGeometry.h [hunk 2]
```diff
@@ -337,7 +337,7 @@ struct WgVertexBufferInd {
         float cosine = dot(nrm, nrm1);
         float angle = std::acos(dot(nrm1, -nrm2));
         float miterRatio = 1.0f / (std::sin(angle) * 0.5f);
-        if (miterRatio <= miterLimit) {
+        if ((miterRatio <= miterLimit) && (!tvg::zero(cosine)) && (!tvg::zero(angle))) {
```
Comment (mgrudzinska) [inline src/renderer/wg_engine/tvgWgGeometry.h:340]: I meant after l.338 to avoid division by 0 while miterRatio is calculated

I confirm, that now the error is gone

Comment (SergeyLebedkin) [reply src/renderer/wg_engine/tvgWgGeometry.h:340]: Lets keep it as is... I dont want to overcomlicate the logic in this place

**Review by mgrudzinska — APPROVED** [review]
